### PR TITLE
Resolves #2423

### LINKED ISSUE #2423: Add-DnsServerResourceRecord fails to support 2048-bit DKIM records
A note about this tool limiting TXT records to 254 characters might be useful. I lost hours to trying to find a solution but I don’t think there is one. The RFC defines TXT strings as being allowed up to 255 characters and you can have multiple strings. Microsoft knows this, at least in terms of the DNS UI tool supporting 255 characters and multiple strings. If an organisation wishes to add a 2048-bit DKIM signature to 160 zones then this tool creates a severe handicap for them. Thanks.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 64226e08-8175-d9b6-1272-a4e016413fb9
* Version Independent ID: fcb077f7-9145-7727-21c6-8c4ed7fda8fd
* Content: [Add-DnsServerResourceRecord (DnsServer)](https://docs.microsoft.com/en-us/powershell/module/dnsserver/add-dnsserverresourcerecord?view=windowsserver2019-ps)
* Content Source: [docset/winserver2019-ps/dnsserver/Add-DnsServerResourceRecord.md](https://github.com/MicrosoftDocs/windows-powershell-docs/blob/master/docset/winserver2019-ps/dnsserver/Add-DnsServerResourceRecord.md)
* Product: **w10**
* Technology: ****
* GitHub Login: @Kateyanne
* Microsoft Alias: **v-kaunu**

### PR DESCRIPTION
**Pull request by velkovb**
Add info about character limit in TXT records
https://github.com/MicrosoftDocs/windows-powershell-docs/issues/2423

### CODE REVIEW

File: docset/winserver2012-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 1]
```diff
@@ -237,6 +237,19 @@ PS C:\>Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Dom
 This command adds a service locator (SRV) resource record for the _sip service on port 5060 with a weight and priority of 0 to the contoso.com domain.
 The host that offers the service is sipserver1.contoso.com.
 
+### Example 8: Add multiline TXT resource record
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/dnsserver/Add-DnsServerResourceRecord.md:240]:
```suggestion
### Example 8: Add a multiline TXT resource record
```

File: docset/winserver2012-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 2]
```diff
@@ -237,6 +237,19 @@ PS C:\>Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Dom
...
+$dkim2 = "v=DKIM1;k=rsa;p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy6OAXCmjYT823gq+DXRjXdsypt7iepfl4pkvLRVN8wRwoND2Fk2aVlG+CitAeJ0nqWn7JAPjoTXpFtHnOWMN7ay/atQd+DcLLHfJkpRvsYSDQ1jkI2s7CkWF6G+nwLGJcNFndOdB8oawpppyESE7+DiZae8bDicaTK8oPU0J7iogeZ1fgvmutwNtNzZHiSgwF9euCiX6lTmGe+0oZ+gRUJnUmZevh//IZ+NyDkRV2kPxQBtM8brHUpRL1c11q/CA0kC6C3ku+Pqmf6A8CGT+qvlCeQ2lVqlBydQL5UjiixUEwkSrgUEKoKE2Hqw97WrDEJZqngtuqma9hWoAsKVbzwIDAQAB"
+$dkim2_part1 = $dkim2.Substring(0,252)
+
+$dkim2_part2 = $dkim2.Substring(252,$dkim2.Length - 252)
+
+Add-DnsServerResourceRecord -DescriptiveText "$dkim2_part1`r`n$dkim_part2" -Name sea2048._domainkey -Txt -ZoneName $domain -TimeToLive 0:1:0:0
+
+```
+
+This command adds a mutliline TXT resource record. String is split in 252 characters sets and then new line is added in the end which is 2 characters in Windows.
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/dnsserver/Add-DnsServerResourceRecord.md:251]:
```suggestion
This command adds a multiline TXT resource record. The string is split into 252 character sets and the new line, which is two characters in Windows, is added at the end.
```

File: docset/winserver2012r2-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 3]
```diff
@@ -239,6 +239,19 @@ PS C:\>Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Dom
 This command adds a service locator (SRV) resource record for the _sip service on port 5060 with a weight and priority of 0 to the contoso.com domain.
 The host that offers the service is sipserver1.contoso.com.
 
+### Example 8: Add multiline TXT resource record
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/dnsserver/Add-DnsServerResourceRecord.md:242]:
```suggestion
### Example 8: Add a multiline TXT resource record
```

File: docset/winserver2012r2-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 4]
```diff
@@ -239,6 +239,19 @@ PS C:\>Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Dom
...
+$dkim2 = "v=DKIM1;k=rsa;p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy6OAXCmjYT823gq+DXRjXdsypt7iepfl4pkvLRVN8wRwoND2Fk2aVlG+CitAeJ0nqWn7JAPjoTXpFtHnOWMN7ay/atQd+DcLLHfJkpRvsYSDQ1jkI2s7CkWF6G+nwLGJcNFndOdB8oawpppyESE7+DiZae8bDicaTK8oPU0J7iogeZ1fgvmutwNtNzZHiSgwF9euCiX6lTmGe+0oZ+gRUJnUmZevh//IZ+NyDkRV2kPxQBtM8brHUpRL1c11q/CA0kC6C3ku+Pqmf6A8CGT+qvlCeQ2lVqlBydQL5UjiixUEwkSrgUEKoKE2Hqw97WrDEJZqngtuqma9hWoAsKVbzwIDAQAB"
+$dkim2_part1 = $dkim2.Substring(0,252)
+
+$dkim2_part2 = $dkim2.Substring(252,$dkim2.Length - 252)
+
+Add-DnsServerResourceRecord -DescriptiveText "$dkim2_part1`r`n$dkim_part2" -Name sea2048._domainkey -Txt -ZoneName $domain -TimeToLive 0:1:0:0
+
+```
+
+This command adds a mutliline TXT resource record. String is split in 252 characters sets and then new line is added in the end which is 2 characters in Windows.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/dnsserver/Add-DnsServerResourceRecord.md:253]:
```suggestion
This command adds a multiline TXT resource record. The string is split into 252 character sets and the new line, which is two characters in Windows, is added at the end.
```

File: docset/winserver2016-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 5]
```diff
@@ -268,6 +268,19 @@ PS C:\> Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Do
 This command adds a service locator (SRV) resource record for the _sip service on port 5060 with a weight and priority of 0 to the contoso.com domain.
 The host that offers the service is sipserver1.contoso.com.
 
+### Example 8: Add multiline TXT resource record
```
Comment (JohanFreelancer9) [inline docset/winserver2016-ps/dnsserver/Add-DnsServerResourceRecord.md:271]:
```suggestion
### Example 8: Add a multiline TXT resource record
```

File: docset/winserver2016-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 6]
```diff
@@ -268,6 +268,19 @@ PS C:\> Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Do
...
+$dkim2 = "v=DKIM1;k=rsa;p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy6OAXCmjYT823gq+DXRjXdsypt7iepfl4pkvLRVN8wRwoND2Fk2aVlG+CitAeJ0nqWn7JAPjoTXpFtHnOWMN7ay/atQd+DcLLHfJkpRvsYSDQ1jkI2s7CkWF6G+nwLGJcNFndOdB8oawpppyESE7+DiZae8bDicaTK8oPU0J7iogeZ1fgvmutwNtNzZHiSgwF9euCiX6lTmGe+0oZ+gRUJnUmZevh//IZ+NyDkRV2kPxQBtM8brHUpRL1c11q/CA0kC6C3ku+Pqmf6A8CGT+qvlCeQ2lVqlBydQL5UjiixUEwkSrgUEKoKE2Hqw97WrDEJZqngtuqma9hWoAsKVbzwIDAQAB"
+$dkim2_part1 = $dkim2.Substring(0,252)
+
+$dkim2_part2 = $dkim2.Substring(252,$dkim2.Length - 252)
+
+Add-DnsServerResourceRecord -DescriptiveText "$dkim2_part1`r`n$dkim_part2" -Name sea2048._domainkey -Txt -ZoneName $domain -TimeToLive 0:1:0:0
+
+```
+
+This command adds a mutliline TXT resource record. String is split in 252 characters sets and then new line is added in the end which is 2 characters in Windows.
```
Comment (JohanFreelancer9) [inline docset/winserver2016-ps/dnsserver/Add-DnsServerResourceRecord.md:282]:
```suggestion
This command adds a multiline TXT resource record. The string is split into 252 character sets and the new line, which is two characters in Windows, is added at the end.
```

File: docset/winserver2012-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 7]
```diff
@@ -237,6 +237,19 @@ PS C:\>Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Dom
...
 
+### Example 8: Add multiline TXT resource record
+```powershell
+$dkim2 = "v=DKIM1;k=rsa;p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy6OAXCmjYT823gq+DXRjXdsypt7iepfl4pkvLRVN8wRwoND2Fk2aVlG+CitAeJ0nqWn7JAPjoTXpFtHnOWMN7ay/atQd+DcLLHfJkpRvsYSDQ1jkI2s7CkWF6G+nwLGJcNFndOdB8oawpppyESE7+DiZae8bDicaTK8oPU0J7iogeZ1fgvmutwNtNzZHiSgwF9euCiX6lTmGe+0oZ+gRUJnUmZevh//IZ+NyDkRV2kPxQBtM8brHUpRL1c11q/CA0kC6C3ku+Pqmf6A8CGT+qvlCeQ2lVqlBydQL5UjiixUEwkSrgUEKoKE2Hqw97WrDEJZqngtuqma9hWoAsKVbzwIDAQAB"
+$dkim2_part1 = $dkim2.Substring(0,252)
+
+$dkim2_part2 = $dkim2.Substring(252,$dkim2.Length - 252)
+
+Add-DnsServerResourceRecord -DescriptiveText "$dkim2_part1`r`n$dkim_part2" -Name sea2048._domainkey -Txt -ZoneName $domain -TimeToLive 0:1:0:0
+
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/dnsserver/Add-DnsServerResourceRecord.md:248]:
```suggestion
```

File: docset/winserver2012r2-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 8]
```diff
@@ -239,6 +239,19 @@ PS C:\>Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Dom
...
 
+### Example 8: Add multiline TXT resource record
+```powershell
+$dkim2 = "v=DKIM1;k=rsa;p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy6OAXCmjYT823gq+DXRjXdsypt7iepfl4pkvLRVN8wRwoND2Fk2aVlG+CitAeJ0nqWn7JAPjoTXpFtHnOWMN7ay/atQd+DcLLHfJkpRvsYSDQ1jkI2s7CkWF6G+nwLGJcNFndOdB8oawpppyESE7+DiZae8bDicaTK8oPU0J7iogeZ1fgvmutwNtNzZHiSgwF9euCiX6lTmGe+0oZ+gRUJnUmZevh//IZ+NyDkRV2kPxQBtM8brHUpRL1c11q/CA0kC6C3ku+Pqmf6A8CGT+qvlCeQ2lVqlBydQL5UjiixUEwkSrgUEKoKE2Hqw97WrDEJZqngtuqma9hWoAsKVbzwIDAQAB"
+$dkim2_part1 = $dkim2.Substring(0,252)
+
+$dkim2_part2 = $dkim2.Substring(252,$dkim2.Length - 252)
+
+Add-DnsServerResourceRecord -DescriptiveText "$dkim2_part1`r`n$dkim_part2" -Name sea2048._domainkey -Txt -ZoneName $domain -TimeToLive 0:1:0:0
+
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/dnsserver/Add-DnsServerResourceRecord.md:250]:
```suggestion
```

File: docset/winserver2016-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 9]
```diff
@@ -268,6 +268,19 @@ PS C:\> Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Do
...
+```powershell
+$dkim2 = "v=DKIM1;k=rsa;p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy6OAXCmjYT823gq+DXRjXdsypt7iepfl4pkvLRVN8wRwoND2Fk2aVlG+CitAeJ0nqWn7JAPjoTXpFtHnOWMN7ay/atQd+DcLLHfJkpRvsYSDQ1jkI2s7CkWF6G+nwLGJcNFndOdB8oawpppyESE7+DiZae8bDicaTK8oPU0J7iogeZ1fgvmutwNtNzZHiSgwF9euCiX6lTmGe+0oZ+gRUJnUmZevh//IZ+NyDkRV2kPxQBtM8brHUpRL1c11q/CA0kC6C3ku+Pqmf6A8CGT+qvlCeQ2lVqlBydQL5UjiixUEwkSrgUEKoKE2Hqw97WrDEJZqngtuqma9hWoAsKVbzwIDAQAB"
+$dkim2_part1 = $dkim2.Substring(0,252)
+
+$dkim2_part2 = $dkim2.Substring(252,$dkim2.Length - 252)
+
+Add-DnsServerResourceRecord -DescriptiveText "$dkim2_part1`r`n$dkim_part2" -Name sea2048._domainkey -Txt -ZoneName $domain -TimeToLive 0:1:0:0
+
+```
+
```
Comment (JohanFreelancer9) [inline docset/winserver2016-ps/dnsserver/Add-DnsServerResourceRecord.md:281]:
```suggestion
```

File: docset/winserver2016-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 10]
```diff
@@ -268,6 +268,19 @@ PS C:\> Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Do
...
 
+### Example 8: Add multiline TXT resource record
+```powershell
+$dkim2 = "v=DKIM1;k=rsa;p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy6OAXCmjYT823gq+DXRjXdsypt7iepfl4pkvLRVN8wRwoND2Fk2aVlG+CitAeJ0nqWn7JAPjoTXpFtHnOWMN7ay/atQd+DcLLHfJkpRvsYSDQ1jkI2s7CkWF6G+nwLGJcNFndOdB8oawpppyESE7+DiZae8bDicaTK8oPU0J7iogeZ1fgvmutwNtNzZHiSgwF9euCiX6lTmGe+0oZ+gRUJnUmZevh//IZ+NyDkRV2kPxQBtM8brHUpRL1c11q/CA0kC6C3ku+Pqmf6A8CGT+qvlCeQ2lVqlBydQL5UjiixUEwkSrgUEKoKE2Hqw97WrDEJZqngtuqma9hWoAsKVbzwIDAQAB"
+$dkim2_part1 = $dkim2.Substring(0,252)
+
+$dkim2_part2 = $dkim2.Substring(252,$dkim2.Length - 252)
+
+Add-DnsServerResourceRecord -DescriptiveText "$dkim2_part1`r`n$dkim_part2" -Name sea2048._domainkey -Txt -ZoneName $domain -TimeToLive 0:1:0:0
+
```
Comment (JohanFreelancer9) [inline docset/winserver2016-ps/dnsserver/Add-DnsServerResourceRecord.md:279]:
```suggestion
```

File: docset/winserver2019-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 11]
```diff
@@ -268,6 +268,19 @@ PS C:\> Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Do
 This command adds a service locator (SRV) resource record for the _sip service on port 5060 with a weight and priority of 0 to the contoso.com domain.
 The host that offers the service is sipserver1.contoso.com.
 
+### Example 8: Add multiline TXT resource record
```
Comment (JohanFreelancer9) [inline docset/winserver2019-ps/dnsserver/Add-DnsServerResourceRecord.md:271]:
```suggestion
### Example 8: Add a multiline TXT resource record
```

File: docset/winserver2019-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 12]
```diff
@@ -268,6 +268,19 @@ PS C:\> Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Do
...
+```powershell
+$dkim2 = "v=DKIM1;k=rsa;p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy6OAXCmjYT823gq+DXRjXdsypt7iepfl4pkvLRVN8wRwoND2Fk2aVlG+CitAeJ0nqWn7JAPjoTXpFtHnOWMN7ay/atQd+DcLLHfJkpRvsYSDQ1jkI2s7CkWF6G+nwLGJcNFndOdB8oawpppyESE7+DiZae8bDicaTK8oPU0J7iogeZ1fgvmutwNtNzZHiSgwF9euCiX6lTmGe+0oZ+gRUJnUmZevh//IZ+NyDkRV2kPxQBtM8brHUpRL1c11q/CA0kC6C3ku+Pqmf6A8CGT+qvlCeQ2lVqlBydQL5UjiixUEwkSrgUEKoKE2Hqw97WrDEJZqngtuqma9hWoAsKVbzwIDAQAB"
+$dkim2_part1 = $dkim2.Substring(0,252)
+
+$dkim2_part2 = $dkim2.Substring(252,$dkim2.Length - 252)
+
+Add-DnsServerResourceRecord -DescriptiveText "$dkim2_part1`r`n$dkim_part2" -Name sea2048._domainkey -Txt -ZoneName $domain -TimeToLive 0:1:0:0
+
```
Comment (JohanFreelancer9) [inline docset/winserver2019-ps/dnsserver/Add-DnsServerResourceRecord.md:279]:
```suggestion
```

File: docset/winserver2019-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 13]
```diff
@@ -268,6 +268,19 @@ PS C:\> Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Do
...
+```
+
+This command adds a mutliline TXT resource record. String is split in 252 characters sets and then new line is added in the end which is 2 characters in Windows.
```
Comment (JohanFreelancer9) [inline docset/winserver2019-ps/dnsserver/Add-DnsServerResourceRecord.md:282]:
```suggestion
This command adds a multiline TXT resource record. The string is split into 252 character sets and the new line, which is two characters in Windows, is added at the end.
```

File: docset/winserver2022-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 14]
```diff
@@ -268,6 +268,19 @@ PS C:\> Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Do
 This command adds a service locator (SRV) resource record for the _sip service on port 5060 with a weight and priority of 0 to the contoso.com domain.
 The host that offers the service is sipserver1.contoso.com.
 
+### Example 8: Add multiline TXT resource record
```
Comment (JohanFreelancer9) [inline docset/winserver2022-ps/dnsserver/Add-DnsServerResourceRecord.md:271]:
```suggestion
### Example 8: Add a multiline TXT resource record
```

File: docset/winserver2022-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 15]
```diff
@@ -268,6 +268,19 @@ PS C:\> Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Do
...
+```powershell
+$dkim2 = "v=DKIM1;k=rsa;p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy6OAXCmjYT823gq+DXRjXdsypt7iepfl4pkvLRVN8wRwoND2Fk2aVlG+CitAeJ0nqWn7JAPjoTXpFtHnOWMN7ay/atQd+DcLLHfJkpRvsYSDQ1jkI2s7CkWF6G+nwLGJcNFndOdB8oawpppyESE7+DiZae8bDicaTK8oPU0J7iogeZ1fgvmutwNtNzZHiSgwF9euCiX6lTmGe+0oZ+gRUJnUmZevh//IZ+NyDkRV2kPxQBtM8brHUpRL1c11q/CA0kC6C3ku+Pqmf6A8CGT+qvlCeQ2lVqlBydQL5UjiixUEwkSrgUEKoKE2Hqw97WrDEJZqngtuqma9hWoAsKVbzwIDAQAB"
+$dkim2_part1 = $dkim2.Substring(0,252)
+
+$dkim2_part2 = $dkim2.Substring(252,$dkim2.Length - 252)
+
+Add-DnsServerResourceRecord -DescriptiveText "$dkim2_part1`r`n$dkim_part2" -Name sea2048._domainkey -Txt -ZoneName $domain -TimeToLive 0:1:0:0
+
```
Comment (JohanFreelancer9) [inline docset/winserver2022-ps/dnsserver/Add-DnsServerResourceRecord.md:279]:
```suggestion
```

File: docset/winserver2022-ps/dnsserver/Add-DnsServerResourceRecord.md [hunk 16]
```diff
@@ -268,6 +268,19 @@ PS C:\> Add-DnsServerResourceRecord -Srv -Name "sip" -ZoneName "contoso.com" -Do
...
+```
+
+This command adds a mutliline TXT resource record. String is split in 252 characters sets and then new line is added in the end which is 2 characters in Windows.
```
Comment (JohanFreelancer9) [inline docset/winserver2022-ps/dnsserver/Add-DnsServerResourceRecord.md:282]:
```suggestion
This command adds a multiline TXT resource record. The string is split into 252 character sets and the new line, which is two characters in Windows, is added at the end.
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review.

**Review by JohanFreelancer9 — APPROVED** [review]

**Review by dariomws — APPROVED** [review]

**Review by get-itips — APPROVED** [review]

**Review by tseward — APPROVED** [review]

**Comment by v-dihans** [comment]
Hi @velkovb  

Merging with non-blocking feedback.

&nbsp;
**Following are non-blocking issues that should be resolved now or within three business days in another PR:**

docset/winserver2012-ps/dnsserver/Add-DnsServerResourceRecord.md
- Line 168: Fix the link. Markdown shows on staging. Also change the "http" to "https" and remove the "en-us" from the link.
I noticed that the link actually redirects to another url. You might want to update the link.

docset/winserver2012r2-ps/dnsserver/Add-DnsServerResourceRecord.md
- Line 170: Fix the link. Markdown shows on staging. Also change the "http" to "https" and remove the "en-us" from the link.
I noticed that the link actually redirects to another url. You might want to update the link.

***

**Reference**

[Criteria used for pull request reviews](https://review.docs.microsoft.com/en-us/office-authoring-guide/pr-criteria?branch=master)

- The content preview must be clean on staging. It cannot contain any obvious formatting issues.
- With rare exceptions, all instances of "http" are changed to "https" in Microsoft links. 
- Links to pages on docs.microsoft.com sites are coded as locale agnostic, meaning that the URL does not include a locale, such as en-us, en-gb, en-in, or any other locale.

***

Also noticed several existing links in the articles that need to have the "http" changed to "https" and the "en-us" removed. 
Those could be fixed or updated as well.

Thanks

**Pull request merged**